OSDOCS-2922: Added a warning for deleting clusters.

EricPonvelle opened the pull request:
This PR adds a note to a few topics to warn users that they should remove all network configurations before they delete a cluster.

JIRA: https://issues.redhat.com/browse/OSDOCS-2922

PREVIEWS:

~~1. **Revoking access to ROSA (STS)** - https://deploy-preview-41128--osdocs.netlify.app/openshift-rosa/latest/rosa_getting_started_sts/rosa-sts-deleting-access-cluster.html~~
2. **Deleting a cluster** - https://deploy-preview-41128--osdocs.netlify.app/openshift-rosa/latest/rosa_getting_started/rosa-deleting-cluster.html
3. **Deleting cluster (STS)** - https://deploy-preview-41128--osdocs.netlify.app/openshift-rosa/latest/rosa_getting_started_sts/rosa-sts-deleting-cluster.html

mjpytlak commented on modules/rosa-deleting-cluster.adoc:
```diff
@@ -10,7 +10,12 @@ endif::[]
+
+[IMPORTANT]
+====
+You must remove any network configurations, such as VPN configurations and VPC peering, before you delete a cluster. If these configurations remain, then the cluster will not delete properly.
```
```suggestion
You must remove any network configurations, such as VPN configurations and VPC peering, before you delete a cluster. If these configurations remain, then the cluster will not delete properly.
+
ifdef:sts[]
Account-wide IAM roles and policies might be used by other ROSA clusters in the same AWS account. You must only remove the resources if they are not required by other clusters.
endif::sts[]
```

EricPonvelle commented:
@okashi18, so I served this section three different ways. @mjpytlak made some good suggestions on how to show this correctly.

Currently, it will display two important notes on the STS side.
![DeleteCluster-Current](https://user-images.githubusercontent.com/16167833/151583835-226d8e25-ba15-4a3b-bd04-b5ff9e5e8a22.png)

This isn't really bad per se, since we want the user two note both important comments. These are separate warnings that can result in issues if ignored.

Another option that Mike documented in the ticket is to combine the two notes based on context. The first image shows the notes combined on STS:
![DeleteCluster-Combined-STS](https://user-images.githubusercontent.com/16167833/151584398-29b076d2-41e5-4b85-bfdd-684c613ca8a7.png)

Then, the non-STS version:
![DeleteCluster-Combined-non-STS](https://user-images.githubusercontent.com/16167833/151584468-b5f32eb2-b643-4689-8987-d63d3338b94a.png)

Finally, here's what it looks like if I change that second important note into a scarily worded paragraph:
![DeleteCluster-In-Line](https://user-images.githubusercontent.com/16167833/151584598-e4539831-27d3-48c0-8b0a-b5daa2a6b1f0.png)

I am flexible with the way we should document this, though I think the second option is a bit too disrupting. I would prefer the third option, first option, then second. Mike can weigh in on this as well.

mjpytlak commented:
@EricPonvelle As we discussed, I find that doc sets (not just Red Hat) overuse note and important admonitions. To take your 3rd option a step further (which I happen to like best), why not list removing network configurations (1st important) as a prerequisite?

Prerequisites

- Remove any network configurations, such as VPN configurations and VPC peering, before you delete a cluster. If these configurations remain, then the cluster will not delete properly.

And to take this even a step further for STS:

Prerequisites

- Remove any network configurations, such as VPN configurations and VPC peering, before you delete a cluster. If these configurations remain, then the cluster will not delete properly.
- Account-wide IAM roles and policies might be used by other ROSA clusters in the same AWS account. Verify that the resources you are going to delete are not required by other clusters.

Note: That I avoided the scary language all-together. :)

0kashi commented:
@mjpytlak - I like that idea of putting the removal of network configurations in the prereqs.  Though I would keep the note about the account roles.  Though, I do see those twice on the page.  Do we even need the first one?

EricPonvelle commented:
@0kashi and @mjpytlak - thanks for the feedback! I updated the PR to include those changes. Basically, I added a prereq to both the STS and non-STS documentation, and kept the original Important note.

0kashi commented on rosa_getting_started/rosa-deleting-cluster.adoc:
```diff
@@ -7,4 +7,9 @@ toc::[]
+[id="prerequisites_rosa-deleting-cluster"]
+== Prerequisites
+
+* If you have a VPC created by a service, you must remove any additional network configurations or add-on services, such as VPN configurations and VPC peering, before you delete a cluster. If these configurations and services remain, the cluster does not delete properly.
```
Three things:
1. I'd rather it state more explicitly that the managed OpenShift service (ROSA, OSD) created the service.  The way it is now, is more generic and not specific.  Like which service?  Any service that created a vpc?  So let's say, something more concrete like "If the VPC was created by the service..." or "In situations where the service created the VPC..." or "if the service was not deployed into an already existing VPC...".

We can use "{product-title}" instead of "service" to make it even clearer.

> ...you must remove any additional network configurations or add-on services, such as VPN configurations and VPC peering...

2. This give the impression that VPN and VPC peering are add-on services.  Those are example "network configurations". 

3. I would prefer to use "additional services" instead of "add-on" services, since that might lead to confusion with our add-ons. I want to to be clear that we mean other **AWS services** that were added to the VPC.

0kashi commented:
lgtm

maxwelldb commented on modules/rosa-deleting-cluster.adoc:
```diff
@@ -10,7 +10,7 @@ endif::[]
 = Deleting a cluster
 
-You can delete an {product-title} cluster using the `rosa` CLI. 
+You can delete an {product-title} cluster using the `rosa` CLI.
```
```suggestion
You can delete an {product-title} cluster by using the `rosa` CLI.
```
ISG

maxwelldb commented on rosa_getting_started/rosa-deleting-cluster.adoc:
```diff
@@ -7,4 +7,9 @@ toc::[]
+[id="prerequisites_rosa-deleting-cluster"]
+== Prerequisites
+
+* If {product-title} created a VPC, you must remove network configurations, such as VPN configurations and VPC peering, and any additional services added to the VPC before you delete your cluster. If these configurations and services remain, the cluster does not delete properly.
```
This is a little wordy for my taste. Maybe consider a bulleted list?

maxwelldb replied:
"Delete" is a transitive verb. 

Could rewrite as something like: "If these configurations or services remain, the cluster cannot delete itself properly."

I'm also curious about what proper deletion entails. Will deletion fail predictably, and so users might follow a procedure elsewhere to fix things, is there chaos, will something error out somewhere, etc., but that's a whole can of worms. ;)

maxwelldb commented on rosa_getting_started_sts/rosa-sts-deleting-cluster.adoc:
```diff
@@ -7,5 +7,10 @@ toc::[]
+[id="prerequisites_rosa-sts-deleting-cluster"]
+== Prerequisites
+
+* If {product-title} created a VPC, you must remove network configurations, such as VPN configurations and VPC peering, and any additional services added to the VPC before you delete your cluster. If these configurations and services remain, the cluster does not delete properly.
```
Same as other file.

maxwelldb reviewed (commented):
Left some nits and feedback. Nothing major.

maxwelldb commented on rosa_getting_started/rosa-deleting-cluster.adoc:
```diff
@@ -7,4 +7,13 @@ toc::[]
+== Prerequisites
+
+* If {product-title} created a VPC, you must remove the following before you delete your cluster:
+  * network configurations, such as VPN configurations and VPC peering
```
Should "VPC peering" be something like "VPC peering connections" or "VPC peering configurations"?

EricPonvelle replied:
@0kashi - which one should it be?

0kashi replied:
VPC peering connections is fine

maxwelldb commented on rosa_getting_started/rosa-deleting-cluster.adoc:
```diff
@@ -7,4 +7,13 @@ toc::[]
+[id="prerequisites_rosa-deleting-cluster"]
+== Prerequisites
+
+* If {product-title} created a VPC, you must remove the following before you delete your cluster:
```
```suggestion
* If {product-title} created a VPC, you must remove the following [$PLURAL_NOUN] before you delete your cluster:
```

Better for translation, but nothing in particular is coming to mind. :(

EricPonvelle replied:
@maxwelldb This work?

* If {product-title} created a VPC, you must remove the following items from your cluster before you can successfully delete your cluster:

maxwelldb commented on rosa_getting_started/rosa-deleting-cluster.adoc:
```diff
@@ -7,4 +7,13 @@ toc::[]
+== Prerequisites
+
+* If {product-title} created a VPC, you must remove the following before you delete your cluster:
+  * network configurations, such as VPN configurations and VPC peering
```
```suggestion
  * Network configurations, such as VPN configurations and VPC peering
```

maxwelldb commented on rosa_getting_started/rosa-deleting-cluster.adoc:
```diff
@@ -7,4 +7,13 @@ toc::[]
+
+* If {product-title} created a VPC, you must remove the following before you delete your cluster:
+  * network configurations, such as VPN configurations and VPC peering
+  * any additional services added to the VPC
```
```suggestion
  * Any additional services that were added to the VPC[,such as $EXAMPLE_1 and $EXAMPLE_2]
```

EricPonvelle replied:
@0kashi what example services should I include?

0kashi replied:
there are tons of aws services such as other ec2 instances, databases, network objects.  I don't know that we need to enumerate them.

EricPonvelle commented:
/cherry-pick enterprise-4.9

EricPonvelle commented:
/cherry-pick enterprise-4.10
